### DISCUSSION TITLE
Steam release for FreeCAD

### DISCUSSION THREAD

**Discussion by BackwardEcho** [FreeCAD/FreeCAD#18624] [Discussions that will turn in to issues but need refining]
Would it be possible to also release FreeCAD through the digital PC game store Steam? 

I asked in the Off-topic Discord channel and was told it isn't very likely to happen, but then I was told to come here and make a post about it. My reasoning for this request is to use the platform as a means for easily updating the program with Blender as an example. I use Blender through Steam and personally love not having to go back to their website periodically to check for updates as I cannot find an update checker within Blender itself.

**Comment by oursland** [comment]
[Blender on Steam](https://store.steampowered.com/app/365670/Blender/)

[Steamworks](https://partner.steamgames.com/steamdirect) provides instructions on how to get an application on the Steam Store.  While focused on games, they do specifically call out "Animation & Modeling" as an accepted application type, for which I believe FreeCAD certainly qualifies.

In general, Steamworks will request tax information for the purposes of withholding which applies to paid games and applications.  There is a $100 per application fee.  A 30 day waiting period takes place after the application fee and the publication of the first release.  A release requires a review, which takes 1-5 days.  A 'coming soon' store page will need to be established at least two weeks prior to release.

**Comment by oursland** [comment]
@adrianinsaval This may provide a means for cross-platform distribution that provides for a convenient auto-update capabilities, similar to the Windows Store and macOS App Store.

**Comment by yorikvanhavre** [comment]
I have thought about steam for some time too... Indeed it might be interesting for FreeCAD, because it's really easy for users, and it would probably not be too much work to maintain (the biggest part of the work is probably to maintain the page itself, with videos, screenshots, text...) but, obviously FreeCAD has not much to do with gaming, or very, very indirectly. And probably, to be fair and clear with users, it should, somehow, be mentioned in the description text that FreeCAD is not really suited to model for games. So does it make sense at all?

**Reply by oursland** [reply]
I agree that FreeCAD is not necessarily well suited towards gaming.  There exists another CAD package on the Steam Store, but it appears to be intended for low-poly artifacts, which I assume is intended for games: [picoCAD](https://store.steampowered.com/app/2800590/picoCAD/).

Even so, it would be a convenient means for cross-platform distribution.  It could be that we would want to try and have them tell us if they believe it is a good fit instead of us telling ourselves no prematurely.

**Reply by yorikvanhavre** [reply]
> It could be that we would want to try and have them tell us if they believe it is a good fit instead of us telling ourselves no prematurely.

That seems like a good plan indeed. The $100 fee is in any case no problem for the FPA. The remaining question is, who's gonna do it :)

**Reply by BackwardEcho** [reply]
> > It could be that we would want to try and have them tell us if they believe it is a good fit instead of us telling ourselves no prematurely.
> 
> That seems like a good plan indeed. The $100 fee is in any case no problem for the FPA. The remaining question is, who's gonna do it :)

I don't think [Wallpaper Engine](https://store.steampowered.com/app/431960/Wallpaper_Engine/) is gaming related, whether it be design or development. It's just fancy wallpapers for your computer.

"do it", as in fork over the $100 for the attempt or make the store page? I can donate something since it was my idea / request but I am broke as hell, so it might not be much. Can't make the page or start the process as I am not associated with you guys nor would I know how with Steam.

**Reply by oursland** [reply]
@BackwardEcho If this is something that the team wishes to pursue, the [FreeCAD Project Association](https://fpa.freecad.org/) could be asked to provide the $100 to register an application.  If you wish to donate in the future, the FPA is a suitable organization to donate for advancement of FreeCAD.

Yorik's reference to "who's gonna do it" is a question of which developer will take on the role and responsibility for applying, releasing, and maintaining FreeCAD on the Steam Store.  A role I would be willing to assist in.

**Reply by yorikvanhavre** [reply]
I will get the FPA approval for the $100, that's no problem. Indeed the question is more, who is going to build the page, update it, maintain it, add the different packages, reply to users questions, etc... That's no small work.

**Reply by memkiii** [reply]
If it is done, you will, like Blender, have to tolerate & moderate the thousands of ridiculous & often abusive discussion comments from casual downloaders who have no interest in the app, but just want something for free, & refuse to consider that it might involve them having to think rather than just bash buttons, then get angry when someone tells them to read a manual or watch a tutorial. Resulting in many negative reviews by angry teenagers.

**Reply by hengze436493** [reply]
I just saw that blender could be downloaded on Steam, and immediately thought of FreeCAD (^~^).

**Comment by yorikvanhavre** [comment]
Okay vote passed, we can go ahead. I also put @oursland in contact with Blender people so they can give us some feedback

**Reply by tobiasfalk** [reply]
Out of interest, how is it standing on the Steam publication?

**Reply by oursland** [reply]
I am behind on getting that going.  I'm in grant writing mode right now until mid-March, which means it consumed all of my time.

**Reply by BackwardEcho** [reply]
This is a question for whenever you do get to creating the page: Do you plan to just upload the stable builds (such as 1.0) or might you also include the more frequent Dev builds in a Beta branch option that people can opt into?

**Comment by BackwardEcho** [comment]
Has there been any developments? I find myself checking for a Steam page every so often as it would probably be what gets me to update to a dev build of FC if those are uploaded as a beta branch.

**Reply by BackwardEcho** [reply]
Apologies if you are busy elsewhere, @oursland, I am just wondering how the process is going.

**Reply by oursland** [reply]
I'm sorry, I've been working on getting weekly builds functioning and have not spent time towards other distribution means as of yet.

**Reply by BackwardEcho** [reply]
Take your time, thank you for your work.

**Comment by BackwardEcho** [comment]
@oursland Would this be considered a "feature" for the September cut-off deadline?

**Reply by oursland** [reply]
No.  Release modes are not features for the purposes of a cutoff.

**Reply by BackwardEcho** [reply]
Ah, ok. I was thinking it might be if the effort needed was close to the same as new feature would require or even as just a "new distribution method to go with the new major version" instead of some bug fixes. Thank you for your work on it.

**Comment by BackwardEcho** [comment]
@oursland Has there been any progress on this? Last I knew you were looking into Blender's updating method.

**Comment by samppa3** [comment]
This would be so good if it came to Steam, it would be nice to track the so-called game hours using the calculator to see how long I've been learning CAD drawing!  : )

**Comment by BackwardEcho** [comment]
Welp, with the new Steam Controller release I unearthed my 2015 version to try and actually use it. I don't know what came over me but I decided to try and make a controller layout for FC. To properly use the virtual menu system Steam Input has I needed to add FC as a non-Steam game to get that overlay (which is unfortunately very buggy with FC, or maybe it's just buggy right now in general according to some community posts I saw). So far I think I have it so I can comfortably swivel the camera / model around. Now I plan to make some Action Sets for Sketcher, Part Design, and probably Assembly. 

And because I didn't like the blank default non-Steam game look, I made some cover art for FC. I can share the images and GIMP xcf files when requested. ~~this is how I found out the **CAD** in Free**CAD** is bold.~~

<img width="1206" height="484" alt="image" src="https://github.com/user-attachments/assets/7ef6bf06-a429-48fa-b9ce-415066ab9ea6" />
<img width="953" height="476" alt="image" src="https://github.com/user-attachments/assets/07373704-0bc5-4485-8d34-6a44e8d22d83" />
<img width="2170" height="972" alt="image" src="https://github.com/user-attachments/assets/d7e4744a-bebd-49d8-9279-ff68dcc813ed" />
<img width="823" height="1271" alt="image" src="https://github.com/user-attachments/assets/71425bb2-49ac-4f15-8b6b-8e8cd80b775b" />


I've been told I am _extra_...

**Reply by BackwardEcho** [reply]
If anyone cares, it is actually viable... provided the virtual menu overlays work. I don't know why it was happening, but my overlay system was downright unusable. Steam Input allows you to create a "Virtual Menu", but it wont refresh on screen unless something else on the controller was being fiddled with. Aside from that, this model was 90 to 95% made using only the controller and the Part Design / Sketcher Workbench. The other % was either finding what broke or typing.

Makes me hope for someone to figure out how to utilize the Gyro input as a space mouse.


<img width="2560" height="1394" alt="image" src="https://github.com/user-attachments/assets/c7210e8e-9efa-42f4-9acb-ba62eacf0449" />
<img width="2560" height="1389" alt="image" src="https://github.com/user-attachments/assets/480f9f70-97b1-4c93-be7f-4f754d343c34" />
<img width="1399" height="909" alt="image" src="https://github.com/user-attachments/assets/f44891dc-cb3c-4fe5-a88e-e0d355762fef" />

**Reply by hengze436493** [reply]
这是我用安卓手机运行的 FreeCAD

        (/^▽^)/

![Screenshot_20260727_150832_com_fct_tiny_VncActivity.jpg](https://github.com/user-attachments/assets/e2dc6dca-b9c0-4891-889a-023aad6ec07c)

下面是用安卓敲 Linux 命令实现过程：
（先在 github 上下载并安装 tinycomputer，然后打开里面模拟的 Debian ）。

chmod +x FreeCAD.AppImage；
./FreeCAD.AppImage --appimage-extract；

嘿嘿嘿 (*^▽^*)，这个方法，我已经用了半年多了（刚开始，软件源里只有 0.20 版本，所以强行提取运行祂）。
